File: wlr/OGUR/OGUR/Strategies/AttackStrategy.cs
using OGUR.Creatures;
using OGUR.Skills;

namespace OGUR.Strategies
{
    public class AttackStrategy : IStrategy
    {
        public AttackStrategy(ICreature parent,params int[] targetTypes)
            : base(parent)
        {
            foreach (var targetType in targetTypes)
            {
                _targets.AddTargetTypes(targetType);
            }
        }

        public override void Act()
        {
            if (AbleToMove())
            {
                if (SkillFactory.IsSkill(_parent.GetActiveSkillName(), AnimationType.RANGED))
                {
                    _parent.SetSkillVector(CalculateTargetVector(_parent.GetLocation(), opponent.GetLocation()));
                    if (_parent.GetSkillVector().GridX != 0 || _parent.GetSkillVector().GridY != 0)
                    {
                        _parent.UseActiveSkill();
                    }
                }
                if (targetPath.HasMoves())
                {
                    nextMove.Copy(targetPath.GetNextMove());
                    _parent.MoveTo(nextMove);
                }
            }
        }
    }
}

File: proj/src/aigilas/strategies/impl/AttackStrategy.java
package aigilas.strategies.impl;

import aigilas.creatures.ICreature;
import aigilas.skills.AnimationType;
import aigilas.skills.SkillLogic;
import aigilas.strategies.IStrategy;
import aigilas.strategies.Strategy;
import spx.bridge.ActorType;

public class AttackStrategy extends IStrategy {
    private int _skillCooldown = 0;
    private int _skillCooldownMax = 10;

    public AttackStrategy(ICreature parent, ActorType... targetTypes)

    {
        super(parent, Strategy.Attack);

        for (ActorType targetType : targetTypes) {
            _targets.addTargetTypes(targetType);
        }
    }

    @Override
    public void act() {
        if (AbleToMove()) {
            _skillCooldown--;
            if (_skillCooldown <= 0) {
                _parent.cycleActiveSkill(1);
                if (SkillLogic.isSkill(_parent.getActiveSkill(), AnimationType.RANGED)) {
                    if (opponent != null) {
                        _parent.setSkillVector(CalculateTargetVector(_parent.getLocation(), opponent.getLocation()));
                    }
                    if (_parent.getSkillVector().GridX != 0 || _parent.getSkillVector().GridY != 0) {
                        _parent.useActiveSkill();
                    }
                } else {
                    _parent.useActiveSkill();
                }
                _skillCooldown = _skillCooldownMax;
            }
            if (targetPath.hasMoves()) {
                nextMove.copy(targetPath.getNextMove());
                _parent.moveTo(nextMove);
            }
        }
    }
}
